### PR TITLE
Add sub_appid

### PR DESCRIPTION
**Pull request by yelizariev**
你好!
It's needed for mini program or in-app payments

I added it only for orders and don't really sure that it works. I'd like to get your opinion about the general idea.

Feel free to reply in Chinese

### CODE REVIEW

**Comment by messense** [comment]
Can you provide the API documentation link of WeChat Pay?

**Comment by sinomate** [comment]
https://pay.weixin.qq.com/wiki/doc/api/wxa/wxa_sl_api.php?chapter=9_1

File: wechatpy/pay/__init__.py [hunk 1]
```diff
@@ -69,8 +69,9 @@ def __new__(cls, *args, **kwargs):
         return self
 
     def __init__(self, appid, api_key, mch_id, sub_mch_id=None,
-                 mch_cert=None, mch_key=None, timeout=None, sandbox=False):
+                 mch_cert=None, mch_key=None, sub_appid=None, timeout=None, sandbox=False):
```
Comment (messense) [inline wechatpy/pay/__init__.py]: Add `sub_appid` to the end, we  don't want to break any code that are using positional args.

**Review by messense — CHANGES_REQUESTED** [review]
I think this is the right way. Some other APIs need to be updated as well.

**Comment by yelizariev** [comment]
unified order has sub_appid marked as required: https://pay.weixin.qq.com/wiki/doc/api/wxa/wxa_sl_api.php?chapter=9_1#

Some explanation about sub_appid for mini program is here: https://pay.weixin.qq.com/wiki/doc/api/wxa/wxa_sl_api.php?chapter=7_10&index=1

File: wechatpy/pay/__init__.py (hunk 1 above)
Comment (yelizariev) [reply wechatpy/pay/__init__.py]: done

**Comment by yelizariev** [comment]
@messense do you think I need to update all methods? I.e. basically all methods that already uses ``self.appid``:


```
 find -iname "*.py" | xargs grep "'appid': self.appid" -A15 | grep "self._post"
./pay/api/redpack.py-        return self._post('mmpaymkttransfers/gethbinfo', data=data)
./pay/api/order.py-        return self._post('pay/orderquery', data=data)
./pay/api/order.py-        return self._post('pay/closeorder', data=data)
./pay/api/order.py-        return self._post('secapi/pay/reverse', data=data)
./pay/api/transfer.py-        return self._post('mmpaymkttransfers/gettransferinfo', data=data)
./pay/api/tools.py-        return self._post('tools/shorturl', data=data)
./pay/api/tools.py-        return self._post('pay/downloadbill', data=data)
./pay/api/tools.py-        return self._post('pay/downloadfundflow', data=data)
./pay/api/tools.py-        return self._post('tools/authcodetoopenid', data=data)
./pay/api/refund.py-        return self._post('secapi/pay/refund', data=data)
./pay/api/refund.py-        return self._post('pay/refundquery', data=data)
./pay/api/coupon.py-        return self._post('mmpaymkttransfers/send_coupon', data=data)
./pay/api/coupon.py-        return self._post('mmpaymkttransfers/query_coupon_stock', data=data)
./pay/api/coupon.py-        return self._post('promotion/query_coupon', data=data)
./pay/api/micropay.py-        return self._post('pay/micropay', data=data)
./pay/api/micropay.py-        return self._post('pay/orderquery', data=data)

```

**Comment by messense** [comment]
I guess so, as long as the doc says it accepts the `sub_appid` parameter.

**Comment by yelizariev** [comment]
ok

File: wechatpy/pay/api/order.py [hunk 2]
```diff
@@ -72,6 +74,7 @@ def create(self, trade_type, body, total_fee, notify_url, client_ip=None,
             'limit_pay': limit_pay,
             'product_id': product_id,
             'openid': user_id,
+            'sub_openid': sub_user_id,
```
[the remark this answers is not in the document]
Comment (yelizariev) [reply wechatpy/pay/api/order.py:77]: Fixed

**Comment by yelizariev** [comment]
I made updates. I hope my Chinese is not too bad :smiley:

File: wechatpy/pay/__init__.py [hunk 3]
```diff
@@ -28,7 +28,8 @@ class WeChatPay(object):
     微信支付接口
 
     :param appid: 微信公众号 appid
-    :param api_key: 商户 key
+    :param sub_appid: 当前调起支付的小程序APPID
+    :param api_key: 商户 key,不要在这里使用小程序的密钥
```
Comment (yelizariev) [inline wechatpy/pay/__init__.py:32]: We were very confused on trying to make payment via mini-program work. The idea is that mini-program's api key is not used to make pay requests, while it's used in [sns/jscode2session](https://developers.weixin.qq.com/miniprogram/dev/api/api-login.html)

**Comment by yelizariev** [comment]
I made updates. I hope my Chinese is not too bad :smiley:

**Review by messense — APPROVED** [review]
LGTM

**Pull request merged**